src/Pages/ProductListPage.js

```diff
@@ -38,67 +43,82 @@ const ProductList = () => {
         });
     }
 
-    const handleCheckboxChange = (event) =>{
+
+    const handleCheckboxChange = (event) => {
         const arrayOfchosedCategories = chosedCategories.items;
 
         const isItOnTheArray = arrayOfchosedCategories.some(
             category => category === event.target.name
         );
 
-        if(!isItOnTheArray)
+        if(!isItOnTheArray && event.target.checked)
             addCategory(arrayOfchosedCategories, event.target.name);
-        if(isItOnTheArray)
+        if(isItOnTheArray && !event.target.checked)
             removeCategory(arrayOfchosedCategories, event.target.name);
 
+        if(chosedCategories.items.length > 0)
+            setShowResetButton(true)
+
         const filteredProducts = filter();
 
         setFilteredProducts({
             products: filteredProducts
         });
     }
 
-
     //If no category is selected show all products
     useEffect(() => {
 
-        if(filteredProducts.products.length === 0)
+        if(filteredProducts.products.length === 0){
             setFilteredProducts({ products: allProducts });
+            setShowResetButton(false);
+        }
 
     }, [filteredProducts]);
 
-
     useEffect(() => {
         setTimeout(() => {
             setHasTimePassed(true);
         }, 2000);
     }, [])
 
-
-
     return(
         <>
             <H1Header>This is the Product List Page</H1Header>
             <ProductListSection>
                 <CategoriesContainer>
                     <h2>Categories</h2>
-                    <ul>
-                        {Categories.map(item => (
-                            <li key={item.id}>
-                                <label>
-                                    <input name={item.data.name} type='checkbox' onChange={handleCheckboxChange}/>
-                                    {item.data.name}
-                                </label>
-                            </li>
-                        ))}
-                    </ul>
+                    {
+                        areCategoriesLoaded ? null :
+                        
+                        <ul>
+                            {Categories.results.map(item => (
+                                <li key={item.id}>
+                                    <label>
+                                        <input name={item.data.name} type='checkbox' onClick={handleCheckboxChange}/>
+                                        {item.data.name}
+                                    </label>
+                                </li>
+                            ))}
+                        </ul>
+                    }
+                    {
+                        showResetButton ? 
+                            <input type="reset" value="Clear filters" onClick={ () => {
+                                setFilteredProducts({ products: [] });
+                                setChosedCategories({ items: [] });
+                                }
+                            }/>
```

Comment:
You should move this to an event handler, rathern than doing the handling in the render function. This is to avoid the recreation of the function each time this component renders.